File: jfingerprintmanager/src/test/java/com/jesusm/jfingerprintmanager/base/BasePresenterTest.java
package com.jesusm.jfingerprintmanager.base;

import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;

import com.jesusm.jfingerprintmanager.authentication.presenter.FingerprintAuthenticationDialogPresenter;
import com.jesusm.jfingerprintmanager.base.hardware.FingerprintHardware;

import org.mockito.Mock;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class BasePresenterTest {

    @Mock
    protected FingerprintAuthenticationDialogPresenter.View mockView;
    @Mock
    private FingerprintHardware mockFingerprintHardware;

    protected FingerprintAuthenticationDialogPresenter createPresenter(FingerprintAuthenticationDialogPresenter.View view) {
        return createPresenter(view, false, true);
    }

    protected FingerprintAuthenticationDialogPresenter createPresenter(FingerprintAuthenticationDialogPresenter.View view,
                                                                     boolean newFingerprintEnrolled, boolean fingerprintAvailable) {
        FingerprintAuthenticationDialogPresenter presenter = new FingerprintAuthenticationDialogPresenter(view);

        if (newFingerprintEnrolled) {
            presenter.newFingerprintEnrolled();
        }

        when(mockFingerprintHardware.isFingerprintAuthAvailable()).thenReturn(fingerprintAvailable);
        presenter.setFingerprintHardware(mockFingerprintHardware, mock(FingerprintManagerCompat.CryptoObject.class));

        return presenter;
    }
}

File: kfingerprintmanager/src/test/kotlin/com/jesusm/kfingerprintmanager/base/BasePresenterTest.kt
package com.jesusm.kfingerprintmanager.base


import android.support.v4.hardware.fingerprint.FingerprintManagerCompat
import com.jesusm.kfingerprintmanager.authentication.presenter.FingerprintAuthenticationDialogPresenter
import com.jesusm.kfingerprintmanager.base.hardware.FingerprintHardware
import com.nhaarman.mockito_kotlin.mock
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when` as _when

open class BasePresenterTest {
    val mockView: FingerprintAuthenticationDialogPresenter.View = mock()

    val mockFingerprintHardware: FingerprintHardware = mock()

    fun createPresenter(view: FingerprintAuthenticationDialogPresenter.View): FingerprintAuthenticationDialogPresenter {
        return createPresenter(view, false, true)
    }

    fun createPresenter(view: FingerprintAuthenticationDialogPresenter.View,
                        newFingerprintEnrolled: Boolean, fingerprintAvailable: Boolean): FingerprintAuthenticationDialogPresenter {
        val presenter = FingerprintAuthenticationDialogPresenter(view)

        if (newFingerprintEnrolled) {
            presenter.newFingerprintEnrolled()
        }

        _when(mockFingerprintHardware.isFingerprintAuthAvailable()).thenReturn(fingerprintAvailable)

        presenter.setFingerprintHardware(mockFingerprintHardware, mock<FingerprintManagerCompat.CryptoObject>(FingerprintManagerCompat.CryptoObject::class.java))

        return presenter
    }
}